hail/src/main/scala/is/hail/expr/ir/MatrixWriter.scala

```diff
@@ -385,7 +388,413 @@ case class MatrixVCFWriter(
   metadata: Option[VCFMetadata] = None,
   tabix: Boolean = false
 ) extends MatrixWriter {
-  def apply(ctx: ExecuteContext, mv: MatrixValue): Unit = ExportVCF(ctx, mv, path, append, exportType, metadata, tabix)
+  def apply(ctx: ExecuteContext, mv: MatrixValue): Unit = {
+    val appendStr = getAppendHeaderValue(ctx.fs)
+    ExportVCF(ctx, mv, path, appendStr, exportType, metadata, tabix)
```

Comment:
One thing that might be nice here, is to force the lowered execution path here. I'm not sure how to go about that though. @tpoterba thoughts/advice?

Reply:
OK, I think this should look like:
```
  def apply(ctx: ExecuteContext, mv: MatrixValue): Unit = {
    val tv = mv.toTableValue
    val ts = RVDToTableStage(tv.rvd, tv.globals.toEncodedLiteral(ctx.theHailClassLoader))
    CompileAndEvaluate(ctx, lower(..., ts, ...))
  }
```